### PR TITLE
fix(demo): address refresh after save

### PR DESCRIPTION
**Pull request by BrocksiNet**
### Description

closes: https://github.com/shopware/frontends/issues/340

### Type of change
- [X] Bug fix (non-breaking change which fixes an issue)

### Additional context

Passing the success event to the next component to trigger a reload after the address is saved.

### CODE REVIEW

**Comment by kstala** [comment]
@BrocksiNet  I checked it in PR preview and the problem still exist

**Review by kstala — CHANGES_REQUESTED** [review]
I checked it in PR preview and the problem still exist

**Comment by BrocksiNet** [comment]
Thx for testing @kstala. And yes I missed one thing.
Pushed the fix for the "New Address" Button.
I was only testing with the "Edit Address" Button.

:+1:

File: templates/vue-demo-store/pages/account/address.vue [hunk 1]
```diff
@@ -72,7 +79,12 @@ onBeforeMount(async () => {
       {{ $t("account.addressAddNew") }}
     </button>
     <SharedModal :controller="addAddressModalController">
-      <SharedAccountAddressForm @success="addAddressModalController.close" />
+      <SharedAccountAddressForm
+        @success="
+          async () =>
```
Comment (mdanilowicz) [inline templates/vue-demo-store/pages/account/address.vue:84]: Maybe adding it as a function in the script would be better? Only for a better readable. What do you think?

File: templates/vue-demo-store/pages/checkout/index.vue [hunk 2]
```diff
@@ -224,6 +224,18 @@ onMounted(async () => {
   });
 });
 
+const refreshAddresses = async (addressId?: string) => {
```
Comment (mdanilowicz) [inline templates/vue-demo-store/pages/checkout/index.vue:227]: There can be a situation where more than one address will be refreshed, for example, when you add a new address in a separate tab. I think it is better to add a general page loader, and remove `addressId` param from this function

**Review by mdanilowicz — CHANGES_REQUESTED** [review]

File: templates/vue-demo-store/pages/account/address.vue (hunk 1 above)
Comment (BrocksiNet) [reply templates/vue-demo-store/pages/account/address.vue:84]: I had it like that before but then I was thinking why a new function that uses some already existing functions and then I find it better to put it on the spot where the function is executed.

File: templates/vue-demo-store/pages/checkout/index.vue (hunk 2 above)
Comment (BrocksiNet) [reply templates/vue-demo-store/pages/checkout/index.vue]: Improved this

**Review by mdanilowicz — APPROVED** [review]

File: templates/vue-demo-store/pages/checkout/index.vue [hunk 3]
```diff
@@ -218,12 +217,17 @@ onMounted(async () => {
...
     getPaymentMethods(),
   ]).finally(() => {
-    isLoading["shippingAddress"] = false;
     isLoading["shippingMethods"] = false;
     isLoading["paymentMethods"] = false;
   });
 });
 
+const refreshAddresses = async () => {
+  isLoading["shipping-address"], (isLoading["billing-address"] = true);
```
Comment (patzick) [inline templates/vue-demo-store/pages/checkout/index.vue:226]: looks like there is something wrong with the loading here

File: packages/composables/src/cms/useCmsElementImage.ts [hunk 4]
```diff
@@ -18,7 +18,7 @@ export type UseCmsElementImage = {
   containerStyle: ComputedRef<CSSProperties>;
   anchorAttrs: ComputedRef<AnchorHTMLAttributes>;
   imageAttrs: ComputedRef<ImgHTMLAttributes>;
-  imageContainerAttrs: ComputedRef<CSSProperties | ImgHTMLAttributes>;
+  imageContainerAttrs: ComputedRef<{ [k: string]: string }>;
```
Comment (patzick) [inline packages/composables/src/cms/useCmsElementImage.ts:21]: I do not like this change - it's more implicit, and we should aim for explicitly - there should not be just "string: string" - we should know what we can expect from this field

**Review by patzick — CHANGES_REQUESTED** [review]
two things from my side

File: packages/composables/src/cms/useCmsElementImage.ts (hunk 4 above)
Comment (BrocksiNet) [reply packages/composables/src/cms/useCmsElementImage.ts:21]: @mdanilowicz I can agree with that :top:

**Review by patzick — APPROVED** [review]

**Pull request merged**